### PR TITLE
Add Service Discovery Use Case

### PR DESCRIPTION
**Pull request by eddie-rowe**
### Description
This page is being created to add high level service discovery information to the docs.

### Links
Asana Task: https://app.asana.com/0/708546246086263/1201846609169222/f
Google Doc: https://docs.google.com/document/d/1VJXRmOIbSezbIxfXfaiLixLHd7XPYqIk6iv_JSgHGPM/edit

### PR Checklist

* [ ] updated test coverage
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/intro/usecases/what-is-service-discovery.mdx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+Consul’s service discovery capabilities help you discover, track, and monitor the health of services within a network. Consul acts as a single source of truth that allows your services to query and communicate with each other.
+
+You can use Consul with virtual machines (VMs), containers, serverless technologies, or with container orchestration platforms, such as [Nomad](https://www.nomadproject.io/) and Kubernetes. Consul is platform agnostic which makes it a great fit for all environments, including legacy platforms.
+
+Consul is available as a [self-managed](/downloads) project or as a fully managed service mesh solution ([HCP Consul](https://portal.cloud.hashicorp.com/sign-in?utm_source=consul_docs)). HCP Consul enables users to discover and securely connect services without the added operational burden of maintaining a service mesh on their own.
+
+## Next
```
Comment (im2nguyen) [inline website/content/docs/intro/usecases/what-is-service-discovery.mdx:88]:
```suggestion
## Next steps
```

File: website/content/docs/intro/usecases/what-is-service-discovery.mdx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+The service catalog is dynamically updated as new instances of the service are added and legacy or unhealthy service instances are removed. Removed services will no longer participate in the load balancing pool for handling service consumer requests.
+
+![Example diagram of how unhealthy services are removed from the service catalog](/img/what_is_service_discovery_3.png)
+
+## What is Service Discovery in Microservices?
+
+In a microservices application, the set of active service instances changes frequently across a large, dynamic environment. These service instances rely on a service catalog to retrieve the most up-to-date access information from the respective services. A reliable service catalog is especially important for service discovery in microservices to ensure healthy, scalable, and highly responsive application operation.
+
+## What are the two Main Types of Service Discovery?
```
Comment (im2nguyen) [inline website/content/docs/intro/usecases/what-is-service-discovery.mdx:45]:
```suggestion
## What are the two main types of service discovery?
```

Can you make the rest of the headers sentence case too?

File: website/data/docs-nav-data.json [hunk 3]
```diff
@@ -14,6 +14,10 @@
             "path": "intro/usecases",
             "hidden": true
           },
+          {
+            "title": "What is Service Discovery?",
```
Comment (im2nguyen) [inline website/data/docs-nav-data.json:18]:
```suggestion
            "title": "Service Discovery",
```

Can you make the other one (service mesh) similar? I think it fits better in the navbar like this

**Review by im2nguyen — APPROVED** [review]
a couple of nuts but this looks great!

**Comment by eddie-rowe** [comment]
Closing and reopening to trigger pipelines.

**Pull request merged**